src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Visitors/ExpressionTreeRewriter.cs

```diff
@@ -440,10 +440,10 @@ private Expr GenerateBuiltInBinaryOperator(ExprBinOp expr)
                 typeL = convertL;
                 didEnumConversion = true;
             }
-            else if (typeL.IsNullableType() && typeL.StripNubs().isEnumType())
+            else if (typeL is NullableType nubL && nubL.UnderlyingType.isEnumType())
             {
                 Debug.Assert(expr.IsLifted);
-                convertL = GetSymbolLoader().GetTypeManager().GetNullable(typeL.StripNubs().underlyingEnumType());
```

Comment:
>StripNubs [](start = 80, length = 9)

Would it make sense to remove `GetUnderlyingType()`, `StripNubs()` and `StripNubs(bool)` in this PR as well? I'm trying to look at their usage now.